iothub/service/src/Amqp/AmqpClientHelper.cs

```diff
@@ -36,18 +36,13 @@ internal static Exception ToIotHubClientContract(Exception exception)
                         IotHubServiceErrorCode.IotHubUnauthorizedAccess);
 
                 case AmqpException amqpException:
-                    Exception ex = ToIotHubClientContract(amqpException.Error);
-                    if (ex is IotHubServiceException hubEx)
-                    {
-                        // pass amqpException as the inner exception of IotHubServiceException
-                        return new IotHubServiceException(
-                            hubEx.Message,
-                            hubEx.StatusCode,
-                            hubEx.ErrorCode,
-                            string.Empty,
-                            amqpException);
-                    }
-                    return ex;
+                    IotHubServiceException ex = ToIotHubClientContract(amqpException.Error);
```

Comment:
Bummer that we have to create a whole new exception when we have one. Perhaps we pass in the inner to ToIotHubClientContract?

Reply:
Sure, I like this idea!